sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -864,9 +849,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              *
              * However, we support a single column RowType through the SingleColumnTableType.
              */
-            Field field = fields.get(0);
-            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            return new SelectSymbol(relation, singleColumnTableType);
+            DataType innerType = fields.get(0).valueType();
+            SingleColumnTableType dataType = new SingleColumnTableType(innerType);
+            final SelectSymbol.ResultType resultType;
+            if (context.isArrayExpression(node)) {
```

Comment:
Is there really no way for us to end up having an intermediate expression between the array expression and the subquery expression? Otherwise this wouldn't be safe.